eppoclient/rules.go

```diff
@@ -159,34 +179,34 @@ func isOne(attributeValue interface{}, s string) bool {
 	}
 }
 
-func evaluateSemVerCondition(subjectValue *semver.Version, conditionValue *semver.Version, condition condition) bool {
+func evaluateSemVerCondition(subjectValue *semver.Version, conditionValue *semver.Version, condition condition) (bool, error) {
```

Comment:
minor: `evaluateSemVerCondition`, `evaluateNumericCondition`, `promoteInt`, `promoteUint`, `promoteFloat` are internal functions and they expect arguments to be of specific type (and they are properly guarded in our code). There's no legitimate reason for them to return an error, so panic seems appropriate as it signifies a programming error.

Reply:
But we don't want bugs / programming errors on our part to tank the users application--so thinking we should avoid `panics` all together and instead our errors will surface as a top-levle error being returned by the `getAssignment()` methods

Reply:
Let's merge this discussion with https://github.com/Eppo-exp/golang-sdk/pull/54#discussion_r1656756419